src/match-path-async.ts

```diff
@@ -125,48 +167,45 @@ function findFirstExistingPath(
       if (err) {
         return doneCallback(err);
       }
-      const mainFieldName = getPrioritizedMainFieldName(
-        packageJson,
-        mainFields
-      );
-      if (mainFieldName) {
-        const file = path.join(
-          path.dirname(tryPath.path),
-          packageJson[mainFieldName]
-        );
-        fileExists(file, (err2, exists) => {
-          if (err2) {
-            return doneCallback(err2);
-          }
-          if (exists) {
-            // Not sure why we don't just return the full path? Why strip it?
-            return doneCallback(undefined, Filesystem.removeExtension(file));
-          }
-          // Continue with the next path
-          return findFirstExistingPath(
-            tryPaths,
-            readJson,
-            fileExists,
-            doneCallback,
-            index + 1,
-            mainFields
-          );
-        });
-      } else {
-        // This is async code, we need to return in an else-branch otherwise the code still falls through and keeps recursing.
-        // While this might work in general, libraries that use neo-async like Webpack will actually not allow you to call the same callback twice.
-        // An example of where this caused issues: https://github.com/dividab/tsconfig-paths-webpack-plugin/issues/11
```

Comment:
Should we keep this comment somewhere or is it no longer relevant?

Reply:
Good point, I'll include it at the bottom return statement. Thanks for pointing it out!